### PR TITLE
Add Florianópolis (SC) spider

### PR DESCRIPTION
**Pull request by cuducos**
Yay, first lines of code in this awesome project!

Some doubts:

**~~1. About `powers` field~~**

~~Florianópolis seams to have an unique gazette for all powers (for example, in [this issue](http://www.pmf.sc.gov.br/arquivos/diario/pdf/31_05_2016_19.03.05.676c30944d7e7b1879a9ca0143e32de7.pdf) we can see in the table of contents that it covers the executive and the legislative). This I hardcoded `powers=None` – is that ok or is there a better way to record that?~~

**UPDATE** Thanks for the [tip](https://github.com/okfn-brasil/diario-oficial/pull/2#discussion_r179343101), @brunolellis : )

**~~2. About `is_extra_edition` fields~~**

~~I browsed Florianópolis portal a bit and I've found no _extra_ issues so far. Thus I also hardcoded `is_extra_edition=False` – is that ok or is there a better way to ponder on that?~~

**UPDATE** I've found one, now I know how to parse it.

**3. How to know if the spider is working?**

I ran it manually as I haven't seen anything in the logs of `docker-compose run`:

```console:
$ docker-compose run --rm processing bash 
root@processing # cd data_collection/gazette
root@processing # scrapy runspider spiders/sc_florianopolis.py
```

Was there a better way to test it?

### CODE REVIEW

**Comment by buenolucas** [comment]
2) também tive essa dúvida
3) Entra na pasta `data_collection` para ter o comando `crawl` disponível
`/mnt/code/data_collection# scrapy crawl  sc_florianopolis`

Na live o Irio fazia um import no `ipdb` e fazia os debugs direto no console.

**Comment by cuducos** [comment]
Cheers, @brunolellis – fixed :tada:

File: processing/data_collection/gazette/spiders/sc_florianopolis.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        """
+        target = date.today()
+        while target >= self.AVAILABLE_FROM:
+            year, month = str(target.year), str(target.month)
+            data = dict(ano=year, mes=month, passo='1', enviar='')
+            yield FormRequest(url=self.URL, formdata=data, callback=self.parse)
+            target = target + relativedelta(months=1)
+
+    def parse(self, response):
+        """Parse each page. Eahc list all gazettes for a given month.
```
Comment (Irio) [inline processing/data_collection/gazette/spiders/sc_florianopolis.py]: `Eahc` -> `Each`

File: processing/data_collection/gazette/spiders/sc_florianopolis.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+from scrapy import FormRequest, Spider
+
+from gazette.items import Gazette
+
+
+class ScFlorianopolisSpider(Spider):
+    name = 'sc_florianopolis'
+    URL = 'http://www.pmf.sc.gov.br/governo/index.php?pagina=govdiariooficial'
+    MUNICIPALITY_ID = '4205407'
+    AVAILABLE_FROM = date(2009, 6, 1)
```
Comment (Irio) [inline processing/data_collection/gazette/spiders/sc_florianopolis.py]: How about changing this to 2015-01-01? Not sure if we're interested in handling download and parsing issues going further this year.

This is an open discussion on this issue: https://github.com/okfn-brasil/diario-oficial/issues/23

File: processing/data_collection/gazette/spiders/sc_florianopolis.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    def start_requests(self):
+        """The City Hall website publish the gazettes in a page with a form
+        that allow users to browse through different years and months. This
+        form sends requests via POST, so this method emulates a series of these
+        POSTs.
+        @url http://www.pmf.sc.gov.br/governo/index.php?pagina=govdiariooficial
+        @returns requests 1
+        """
+        target = date.today()
+        while target >= self.AVAILABLE_FROM:
```
Comment (Irio) [inline processing/data_collection/gazette/spiders/sc_florianopolis.py]: Cool that you gave preference to start collecting most recent gazettes before going to previous years.

**Comment by Irio** [comment]
@cuducos 
> 3. How to know if the spider is working?

How I personally do it at this moment:

```console
$ docker-compose run processing sh

$ cd data_collection
$ scrapy crawl <spider_name>
```

**Pull request merged**